go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -108,78 +120,71 @@ def update_go_mod(go_mod_content, dependencies)
             }
           end
 
-          SharedHelpers.run_helper_subprocess(
+          body = SharedHelpers.run_helper_subprocess(
             command: NativeHelpers.helper_path,
             env: ENVIRONMENT,
             function: "updateDependencyFile",
             args: { dependencies: deps }
           )
+
+          write_go_mod(body)
         end
 
-        def run_go_get(go_mod_content, go_sum)
-          File.write("go.mod", go_mod_content)
-          File.write("go.sum", go_sum.content) if go_sum
-          File.write("main.go", dummy_main_go)
+        def run_go_get
+          unless Dir.glob("*.go").any?
+            File.write("main.go", "package dummypkg\n")
```

Comment:
Nit: this isn't a problem _yet_...

Once we're doing major version upgrades, we're affecting `**/*.go`, so we'll probably end up with something like `git status ...` (like `bundler`, right?).
I'm 😨 this temp go file will be caught then.

Reply:
Yeah, thanks for calling that out! I was thinking once we start doing that, maybe we can create a file with a random unique identifier here and filter that out when finding the changed files?

Reply:
Could we just remove it in an `ensure` block after calling `go get`?

Reply:
I'm worried that we'll delete actual `main.go` files that way, there's definitely a way to reliably do it, but do you think we should already worry about this now, or can we punt on it until we shell out to git to find changed files?

Reply:
Scratch that, might as well do it now before we forget, I pushed a fix, lmk what you think!